typescript-demo/Dockerfile

```diff
@@ -0,0 +1,57 @@
+FROM cgr.dev/chainguard/node AS base
+
+USER root
+
+# Install dependencies only when needed
+FROM base AS deps
+
+WORKDIR /app
+
+COPY package.json package-lock.json ./
+RUN npm ci
+
+# Rebuild the source code only when needed
+FROM base AS builder
+WORKDIR /app
+COPY --from=deps /app/node_modules ./node_modules
+COPY . .
```

Comment:
Forgive my Node ignorance here - is copying _everything_ required? Although I suppose it's not that big a concern if we're not keeping the image afterwards.

Reply:
Probably not, but in theory everything not in dockerignore should be there.